numba/tests/npyufunc/test_errors.py

```diff
@@ -61,7 +61,7 @@ def check_gufunc_raise(self, **vectorize_args):
         arr = np.array([1, 2, -3, 4], dtype=np.int32)
         out = np.zeros_like(arr)
         with self.assertRaises(ValueError) as cm:
-            f(arr, 2, out)
+            f(arr, int32(2), out)
         # The gufunc bailed out after the error
         self.assertEqual(list(out), [2, 4, 0, 0])
```

Comment:
Hi @sklam, thanks for the review. I've made an attempt to use `ufunc_find_matching_loop`. By using this function rather than calling the ufunc directly, Numba fails in gufuncs compiled for `int32` with `int64` as input, where numpy works just fine. This change will probably crash any code being used nowadays by numba users. Is `ufunc_find_matching_loop` missing something or is there any way to avoid this?

Reply:
Something that I can do is to revert the code to the old approach (call the ufunc directly) and check the TypeError message:

```python
def __call__(self, *args):
    cant_cast_msg = "not supported for the input types, and the inputs" \
                    " could not be safely coerced to any supported types" \
                    " according to the casting rule"
    nonetype_not_callable = "'NoneType' object is not callable"

    try:
        return self.ufunc(*args)
    except TypeError as e:
        if cant_cast_msg in str(e) or nonetype_not_callable in str(e):
            sig = self._get_signature(*args)
            self.add(sig)
            self.build_ufunc()
            return self.ufunc(*args)
        else:
            raise e
```